src/Inkluzitron/Modules/BdsmTestOrg/BdsmModule.cs

```diff
@@ -0,0 +1,240 @@
+                }
+            }
+
+            if (traitsToSearchFor.Count == 0)
+                traitsToSearchFor.UnionWith(Settings.TraitList);
+
+            var relevantItems = DbContext.BdsmTestOrgQuizResults.AsQueryable()
+                .GroupBy(r => r.SubmittedById, (_, results) => results.Max(r => r.ResultId))
+                .Join(
+                    DbContext.DoubleQuizItems.Include(i => i.Parent).Where(i => i.Parent is BdsmTestOrgQuizResult),
+                    x => x,
+                    y => y.Parent.ResultId,
+                    (_, r) => r
+                );
+
+            var resultsDict = new ConcurrentDictionary<string, List<QuizDoubleItem>>();
+            foreach (var trait in traitsToSearchFor)
+            {
+                resultsDict[trait] = relevantItems
+                    .Where(i => i.Key == trait)
+                    .Where(i => i.Value > Settings.TraitReportingThreshold)
+                    .OrderByDescending(row => row.Value)
+                    .ThenByDescending(row => row.Parent.SubmittedAt)
+                    .Take(Settings.MaximumMatchCount)
+                    .ToList();
+            }
+
+            var results = new StringBuilder();
+            foreach (var trait in traitsToSearchFor.OrderBy(x => x, StringComparer.OrdinalIgnoreCase))
+            {
+                string resultsLine;
+
+                var resultsGot = resultsDict.TryGetValue(trait, out var items) && items.Count > 0;
+                if (resultsGot)
+                    resultsLine = string.Join(", ", items.Select(i => $"<@{i.Parent.SubmittedById}> ({i.Value:P0})"));
+                else if (showAllMode)
+                    resultsLine = Settings.NoMatchesMessage;
+                else
+                    continue;
+
+                results.AppendFormat("**{0}**: ", trait);
+                results.AppendLine(resultsLine);
+            }
+
+            if (results.Length == 0)
+                results.Append(Settings.NoMatchesMessage);
+
+            await ReplyAsync(results.ToString(), allowedMentions: new AllowedMentions(AllowedMentionTypes.None));
+        }
+
+        [Command("add")]
+        [Summary("Přidá do databáze výsledek testu.")]
+        public async Task ProcessQuizResultSubmission([Name("<textová forma výsledků>")] params string[] textResults)
+        {
+            var reconstructedMessage = Context.Message.ToString();
+            var testResultMatches = TestResultRegex.Matches(reconstructedMessage);
+            if (testResultMatches.Count == 0)
+            {
+                await ReplyAsync(Settings.InvalidFormatMessage);
+                return;
+            }
+
+            var testResultMatch = testResultMatches.Single();
+            var testResultItems = testResultMatch.Groups["results"].Value;
+            var testResultLink = testResultMatch.Groups["link"].Value;
+
+            if (await DbContext.BdsmTestOrgQuizResults.AsAsyncEnumerable().AnyAsync(r => r.Link == testResultLink))
```

Comment:
```suggestion
            if (await DbContext.BdsmTestOrgQuizResults.AsQueryable().AnyAsync(r => r.Link == testResultLink))
```

Pokud se nepletu, tak `.AsAsyncEnumerable` zařízne SQL dotaz, tudíž se SQL vyhodnotí, že se mají stáhnout všechna data.